### DISCUSSION TITLE
How to make a task depend on a project?

### DISCUSSION THREAD

**Discussion by ofossum** [GothenburgBitFactory/taskwarrior#2782] [Q&A]
Hi,

i just asked myself, how i could tell TW that one task depends on another project.
(As in: `task 12 mod depends:my_project`)
Is that somehow possible?
I couldn't find infos on this.

Thanks in advance! :)

**Comment by smemsh** [comment]
I don't think so, projects are only hierarchy components, really just an attribute of a task, not things in and of themselves that can be depended on.  You would have to make a task that had dependencies on all tasks in a project instead, and then depend on that one.  It's an interesting idea though.

**Comment by Arwalk** [comment]
You can't do it directly, but you can make a task depending on all the task of a projects  in one go by playing a bit with the dom.

Example: here is a starting workspace with 3 tasks in a project, and our "target" task

![image](https://user-images.githubusercontent.com/4776412/173068996-2e7d96ea-9782-410c-96b6-d5238216784c.png)

you can get the ids of a list of task using `_ids` , so with a bit of transformation, you can do something like this

`task mod 4 dep:$(task proj:testproj _ids | perl -p -e 's/\n/,/' | sed 's/,$//')`

Which successfully makes task `4` to suddenly depend to task 1, 2 and 3.

![image](https://user-images.githubusercontent.com/4776412/173069305-4ec9e476-4c77-4160-b488-a6758f0575d9.png)

The main con of this method is that if your project evolves, its not going to be updated automatically.

Other solutions would be to have a "end of project" task in your project, and having this task being the dependency of the task requiring the previous project. That way, only the "end of project" task would need to be updated if your project evolves.
